### PR TITLE
Feature/test coverage

### PR DESCRIPTION
**Pull request by lamorrison**
Adds some tests to `VariablesFetchService`
- Quite verbose comments
- Refactored `fetch_all` so that href gets populated (TODO a test to demonstrate this bug which is now fixed)
- Refactored `fetch_all` so results are streamed as they are complete rather than returning the entire list at the end
- Refactored `rake fetch:variables` which had been fetching each Variable twice
- Added OPENFISCA_SERVER to the Travis environment variables (in .travis.yml)
- Uses Yard documentation conventions in the comments

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,4 +28,5 @@ deploy:
   - restart
 env:
   global:
-    secure: T7+npfKb6EW0G++OP4C91y2WMrMYyOrozeNT1WPCuedd5vKdILSskMz5LBav54F0J6OBbwozH82e0xclMTa23VuOrc9805tEcwtSRdDYb/mIb7F6gEyseqLIk/TZvDfmB5Yrrji0IVX9s5q0OKfVVb0rTrgh87pSRW2YnYAu41k6v2pK/OquixQX0sogblOLfKYprqh4h3Ay6+1IIDmC/lg0XSe+0xRFBZ5FJk0ycStu74xM+qbphIFgzd6q2kChFqRrOporeCgN3WSAFsQd7QneLITN3eucH9AR0OC9Udpuxr8tvAqTfy7Lo3ogj9PTYjD60ObQjcr212d/isSaZd6Zk2wHH35bzsZU3Y5f6vsILiZVJY2rDPkf7A6jMrWSPw8jd/CRizFjedveF9jCy3J2viwNI23y4Ce+sQkc5u0dAV3Z7n5Avy1R/UjyBpRoYb7VhfX1mZTKsVfLPoOxMuvChyOAL4MZGR83N9emHoLFL1r/JsRlqtun/F8i7vsu3OmqpO9ZcexEh+jx1Hqt6qSizNto6bQkKQ5WOul0kfT9U4kIHcKe0EIsUmD2d6CYLXBIYuSitp3/Fq/yX/tUwm9tJED+J0g76g8ItFAznf+7nYe4MSh039ehOKXLW4v+w6624RYfmjCauNHb4HSmwE0n2y8XgLwat7Tj1hwV8Qc=
+    - OPENFISCA_URL=https://api.rules.nz
```
Comment (mischa-s) [inline .travis.yml]: rough, good find :mag:

File: app/services/variables_fetch_service.rb [hunk 2]
```diff
@@ -1,16 +1,36 @@
...
+  #
+  # Returns the loaded Variables, as a streamed enumerable
   def self.fetch_all
-    json_response = of_conn.get('variables').body
-    json_response.keys.each do |name|
-      variable = Variable.find_or_initialize_by(name: name)
+    variables_list.each do |v|
+      # Find or create the Variable model
+      variable = Variable.find_or_initialize_by(name: v.first)
+      # Update the model with attributes retrieved from the server
```
Comment (mischa-s) [inline app/services/variables_fetch_service.rb]: what is this v.first v.second business?

File: app/services/variables_fetch_service.rb [hunk 3]
```diff
@@ -1,16 +1,36 @@
...
+      variable.update(v.second)
+      # Fetch the additional attributes from the server and save to the database
       fetch(variable)
+      yield variable
     end
   end
 
+  # Load the full data of a variable into the database
+  #
+  # The parameter is a Variable containing the short info (only :name is
```
Comment (mischa-s) [inline app/services/variables_fetch_service.rb]: Paramater is a domain specific word in this context, maybe use a synonym?

Comment (lamorrison) [reply app/services/variables_fetch_service.rb]: resolved with overkill - now uses the @param and @return conventions for Yard documentation

**Pull request merged**